src/contracts/core/scdp/facets/SCDPFacet.sol

```diff
@@ -34,6 +34,7 @@ contract SCDPFacet is ISCDPFacet, Modifiers {
         address _collateralAsset,
         uint256 _amount
     ) external payable nonReentrant gate(_account) {
+        cs().onlyUnpaused(_collateralAsset, Enums.Action.SCDPDeposit);
```

Comment:
Lets rather use the `onlyUnpaused` inside the getters, in this case it would be `cs().onlyFeeAccumulatingCollateral`.

Reply:
fixed and also added safety states for swapSCDP.